experimental/packages/opentelemetry-sdk-metrics-base/src/export/MetricReader.ts

```diff
@@ -14,52 +14,123 @@
 import { MetricProducer } from './MetricProducer';
+import { MetricData } from './MetricData';
+import { callWithTimeout } from '../utils';
+
+export type ReaderOptions = {
+  timeoutMillis?: number
+}
+
+export type ReaderCollectionOptions = ReaderOptions;
+
+export type ReaderShutdownOptions = ReaderOptions;
+
+export type ReaderForceFlushOptions = ReaderOptions;
 
 // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricreader
 
+/**
+ * A registered reader of metrics that, when linked to a {@link MetricProducer}, offers global
+ * control over metrics.
+ */
 export abstract class MetricReader {
+  // Tracks the shutdown state.
   private _shutdown = false;
+  // MetricProducer used by this instance.
   private _metricProducer?: MetricProducer;
 
-  constructor(private _exporter: MetricExporter) {}
+  constructor(private readonly _preferredAggregationTemporality = AggregationTemporality.CUMULATIVE) {
+  }
 
+  /**
+   * Set the {@link MetricProducer} used by this instance.
+   *
+   * @param metricProducer
+   */
   setMetricProducer(metricProducer: MetricProducer) {
     this._metricProducer = metricProducer;
+    this.onInitialized();
   }
 
+  /**
+   * Get the {@link AggregationTemporality} preferred by this {@link MetricReader}
+   */
   getPreferredAggregationTemporality(): AggregationTemporality {
-    return this._exporter.getPreferredAggregationTemporality();
+    return this._preferredAggregationTemporality;
   }
 
-  async collect(): Promise<void> {
+  /**
+   * Handle once the SDK has initialized this {@link MetricReader}
+   * Overriding this method is optional.
+   */
+  protected onInitialized(): void {
+    // Default implementation is empty.
+  }
+
+  /**
+   * Handle a shutdown signal by the SDK.
+   *
+   * <p> For push exporters, this should shut down any intervals and close any open connections.
+   * @protected
+   */
+  protected abstract onShutdown(): Promise<void>;
+
+  /**
+   * Handle a force flush signal by the SDK.
+   *
+   * <p> In all scenarios metrics should be collected via {@link collect()}.
+   * <p> For push exporters, this should collect and report metrics.
+   * @protected
+   */
+  protected abstract onForceFlush(): Promise<void>;
+
+  /**
+   * Collect all metrics from the associated {@link MetricProducer}
+   */
+  async collect(options: ReaderCollectionOptions): Promise<MetricData[]> {
     if (this._metricProducer === undefined) {
-      throw new Error('MetricReader is not bound to a MeterProvider');
+      throw new Error('MetricReader is not bound to a MetricProducer');
     }
-    const metrics = await this._metricProducer.collect();
 
-    // errors thrown to caller
-    await this._exporter.export(metrics);
+    // Subsequent invocations to collect are not allowed. SDKs SHOULD return some failure for these calls.
+    if (this._shutdown) {
+      throw new Error('Collection is not allowed after shutdown');
```

Comment:
Do we want to throw an error here? or just warn and return empty?

Reply:
I added this since the spec mentions that when collect is called after shutdown
> SDKs SHOULD return some failure for these calls. 

I think both would be fine, but according to the spec I would expect an Error to be thrown.

Reply:
+1  for a `warn` log, maybe we can stretch "return some failure" to emit a warning, or double check with spec about that

Reply:
I changed it in 2de4fc9 so that `collect()` logs and returns empty on a shut down instance, `forceFlush()` now just logs and returns as well, `shutdown()` already just logged and returned so no changes there. 

Thanks for the feedback :)